staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -608,3 +646,57 @@ func (c *$.type|privatePlural$) Patch(ctx context.Context, name string, pt $.Pat
 	return
 }
 `
+
+var applyTemplate = `
+// Apply takes the given apply declarative configuration, applies it and returns the applied $.resultType|private$.
+func (c *$.type|privatePlural$) Apply(ctx context.Context, $.inputType|private$ *$.applyConfig|raw$, opts $.ApplyOptions|raw$) (result *$.resultType|raw$, err error) {
+	patchOpts := opts.ToPatchOptions()
+	data, err := $.jsonMarshal|raw$($.inputType|private$)
+	if err != nil {
+		return nil, err
+	}
+    name := $.inputType|private$.Name
+	if name == nil {
+		return nil, fmt.Errorf("$.inputType|private$.Name must be provided to Apply")
+	}
+	result = &$.resultType|raw${}
+	err = c.client.Patch($.ApplyPatchType|raw$).
+		$if .namespaced$Namespace(c.ns).$end$
+		Resource("$.type|resource$").
+		Name(*name).
+		VersionedParams(&patchOpts, $.schemeParameterCodec|raw$).
+		Body(data).
+		Do(ctx).
+		Into(result)
+	return
+}
+`
+
+var applyStatusTemplate = `
+// ApplyStatus was generated because the type contains a Status member.
+// Add a +genclient:noStatus comment above the type to avoid generating ApplyStatus().
+func (c *$.type|privatePlural$) ApplyStatus(ctx context.Context, $.inputType|private$ *$.applyConfig|raw$, opts $.ApplyOptions|raw$) (result *$.resultType|raw$, err error) {
```

Comment:
I don't think we've talked about adding "ApplyStatus", but I think I agree it makes sense for controllers to have easy access to that function. This makes things more difficult if someone wants to apply to another subresource (scale?), but it also doesn't make as much sense IMO. So I think that's good.

Reply:
We did sneek this requirement into the KEP at the last minute, @deads2k noticed it was missing.  Functions of the form`<Operation>Status`  exist for update, so this does keep to the client-go convention. If we introduce scale apply support, there is also a convention for that-- one the developer annotates that the scale subresource supports apply, a `ScaleStatus` function would be generated. We will need to update this generator to support that, but most of the building blocks exist, so it shouldn't be terribly difficult to support.